src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -105,12 +108,26 @@ public UnitSpriteSet(String unitType, String factionName, ColorPalette coColors)
 
   }
 
-  private static String getMapUnitSpriteFilename(String unitType, String faction, AnimState state)
+  /**
+   * Attempt to load the sprite for the given unit type, as owned by the specified faction.
+   * If the specified faction has no sprite for that unit, it will try to load it from that faction's
+   * basis instead. If the basis also has no sprite, then it will default to loading the "Thorn" version.
+   * @return The name of the file to load.
+   */
+  private static String getMapUnitSpriteFilename(String unitType, Faction faction, AnimState state)
```

Comment:
I have a quibble with putting this logic at this tier: if e.g. Nova has a unique inf sprite but doesn't have move animations for it, then my space alien will turn into a hunam when he runs around. This is not ideal.

Reply:
...good documentation, though.